gapic/src/main/com/google/gapid/widgets/LoadingIndicator.java

```diff
@@ -148,22 +148,43 @@ public Widget createWidgetWithRefresh(Composite parent) {
 
   /**
    * Widget that shows the loading indicator while loading and is blank once done.
+   * Can optionally show an image when done.
    */
   public class Widget extends Canvas implements Loadable, Repaintable {
-    private boolean loading = false;
+    private Image successImage;
+    private Image failureImage;
+
+    protected boolean loading = false;
+    protected boolean status = false;
 
-    public Widget(Composite parent, boolean showRefresh) {
+    public boolean hasImage = false;
```

Comment:
This should be private and final, however, I don't think the field is necessary.

Reply:
The only loading widget currently in use is the refresh devices, and the validation status. But if someone uses just the loader without refresh or the image, it will cause a null pointer derefencing, which is why I thought this preliminary check might come in handy

Reply:
sorry, i think GH ate one of my comments. The reason I said it was not necessary is because of the comment that got eaten. Below, in the paint handler I was saying to to check each image for null, because `hasImage` is true if either image is not null, so if only a success image was given, but a failure is to be shown, it would fail with an NPE.

I think the code below should be:
```
if (loading) {
  ..
} else {
  Image image = status ? sucessImage : failureImage;
  if (image != null) {
    paint(image, ...);
  }
}
```
Making that change would make `hasImage` obsolete, while supporting various cases of image present/not-present.

Reply:
Done. Updated this in the AGI PR.